### PR TITLE
Added Feature to inspect node by Chef::Platform.windows?

### PR DESCRIPTION
**Pull request by piyushawasthi**
- Added Feature to inspect node by Chef::Platform.windows?
- Re-factor all codes and specs related to Chef::Platform.windows?
- Re-factor code of provider_mapping.rb. 
- Fixed failing specs.

### CODE REVIEW

**Comment by btm** [comment]
@lamont-granquist was this what you had in mind for ChefSpec?

File: chef-config/lib/chef-config/windows.rb [hunk 1]
```diff
@@ -16,13 +16,17 @@
 #
 
 module ChefConfig
-
   def self.windows?
-    if RUBY_PLATFORM =~ /mswin|mingw|windows/
+    if @node
+      platform = @node[:platform] || @node[:platform_family] || @node[:os]
```
Comment (lamont-granquist) [inline chef-config/lib/chef-config/windows.rb]: i don't think you need to do quite this much fallback, on windows you should always have node[:os] and it should always be "windows"

i probably would have shortened this all up to:

```
  def self.windows?
    if @node && @node["os"]
      @node["os"] == "windows"
    else
      RUBY_PLATFORM =~ /mswin|mingw|windows/
    end
end

**Comment by lamont-granquist** [comment]
other than that LGTM

**Comment by btm** [comment]
@piyushawasthi can you do something about the `C:/projects/chef/chef-config/lib/chef-config/windows.rb:20: warning: instance variable @node not initialized` messages spewing in the appveyor output?

**Comment by lamont-granquist** [comment]
`if instance_variable_defined?(:@node) && @node[:os]`?

**Comment by lamont-granquist** [comment]
you might actually want to go in through the accessor though:  `if defined?(node) && node["os"]`

**Comment by piyushawasthi** [comment]
@btm Fixed warning: instance variable @node not initialized messages spewing in the appveyor output?

**Comment by mwrock** [comment]
I approve but the appveyor failure is curious. I'm rerunning the PR to see if it is transient.

**Comment by smurawski** [comment]
@mwrock I'm seeing the same failure running `bundle exec rake spec` locally.

**Comment by btm** [comment]
Replaced by https://github.com/chef/chef/pull/6030

**Pull request closed without merging**